### ISSUE TITLE
Staging Review [Caregivers, File Upload Error Workflow]

### ISSUE THREAD

**Issue by tchrisdyer** [department-of-veterans-affairs/va.gov-team#26697]
## Steps to complete Staging Review: 

- [x] VFS Product Manager: Create this issue and fill in your team and feature name in the title.
- [ ] VFS: Using the Slack Workflow ([User guide](https://depo-platform-documentation.scrollhelp.site/support/Getting-help-from-the-Platform-in-Slack.1439138197.html)), notify Collaboration Cycle team of this Github ticket in #vfs-platform-support. 
- [x] Platform: Schedule meeting with Platform reviewers and requesting team attendees (as listed below)
- [ ] VFS: At least 4 days before scheduled meeting, you must complete list of artifacts below
- [ ] VFS: Verify that access information has been provided for the tool or feature in staging. Do not put staging credentials in your va.gov-team ticket; store or reference them in a .md file in the va.gov-team-sensitive repository.
- [x] Platform: Create feedback tickets and link to the Staging Review ticket (if no feedback, please comment stating that). 
- [ ] VFS: Respond to Platform feedback tickets.
- [ ] VFS: Update this ticket with the product’s launch date after it has gone live.
- [ ] Platform: Update [Platform Collaboration Point Tracker](https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=266151061).
- [ ] VFS: Close this ticket when the review is complete.

## Artifacts
- [product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/teams/caregiver/product-outline.md)
- Staging Link:

> Single file upload form:
> Navigate to: https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/introduction
> You only need to fill out primary or secondary caregiver, you do not have to fill out both
> When you get to Step 4 of 5: Representative document, make sure you answer `Yes. I want to upload my document now. The next page will be the file upload.
> 
> Multiple file upload form:
> Navigate to: https://staging.va.gov/disability/file-disability-claim-form-21-526ez/start
>Answer `No` to `Are you on active duty right now?`
>Answer `Yes` to `Are you filing a new claim or are you disagreeing with a VA decision on an earlier claim?`
> Login. Credentials for vets.gov.user+33@gmail.com worked for me.
> On Step 3 of 5: Supporting evidence, make sure to select Yes on the question Is there any evidence you’d like us to review as part of your claim?
> On the question What type of evidence do you want us to review as part of your claim? select Private medical records
> On the next page, select yes to the question Do you want to upload your private medical records? The next page will be the file upload
-  Test case: [21-526EZ](https://dsvavsp.testrail.io/index.php?/cases/view/5515), [1010CG](https://dsvavsp.testrail.io/index.php?/cases/view/5512)
-  Staging test-user information _(n/a)_
-  Link to content brief _(n/a)_
-  [content source of truth](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/caregivers/1010cg-mvp/10-10CG-application-copy.md) 
-  Link to the Github ticket that shows the work for creating and/or updating any relevant static content pages and entry points for this project
- Release plan: _as this is a small incremental change to an existing message, the product will launch at 100%._
- Product Test Cases: _as this is a small incremental change to an existing message for an intermittent issue, we are primarily testing ad hoc._
- Coverage for Reference: _as this is a small incremental change to an existing message, no major adjustments being made_
- Link to verification that accessibility checks are complete:
  - (preferred) Link to your [TestRail Platform accessibility test plan template](https://dsvavsp.testrail.io/index.php?/suites/view/14&group_by=cases:section_id&group_order=asc).
OR
  - Link to a doc in your product folder that indicates all required checks from the accessibility [staging review prep](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md) doc are complete.
- List of known issues (with links to tickets when applicable)

## Meeting attendees from requesting team
- Product Manager (required): Chris Dyer
- OCTO-DE Product lead (recommended): Ming Ligh
- OCTO-DE Design lead (required): Ryan Thurlwell
- Anyone else from your team who significantly contributed to the artifacts provided (required): Dat Tran, Dené Gabaldón
- VFS Content, IA, accessibility, and QA specialists (if applicable): name(s) Danielle Thierry, Angela Fowler

## Scheduling
What is your preferred time slot per the [Platform Collaboration Meetings Calendar](https://calendar.google.com/calendar/u/0/embed?src=adhocteam.us_4dn3o77gcm5e3vbiedlha96tc0@group.calendar.google.com&ctz=America/New_York)? **July 6 / 11:30am**
 
Please note:
- All times are shown in ET. 
- Available time slots are indicated with [CALENDAR BLOCK]. 
- When you reserve a time slot, the reserved meeting will show up alongside the [CALENDAR BLOCK]. 
- Please select a time slot at least 4 business days from request date.

## Additional information
Please refer to the [Platform Collaboration Cycle README](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/README.md) on Github for more information about the Collaboration Cycle.

**Comment by shiragoodman** [comment]
hi @tchrisdyer. Per our convo on Slack, the Staging Review has been scheduled for 7/8 at 1:30 pm ET. Please ensure all artifacts are ready to review by the end of day Friday 7/2.

**Comment by f1337** [comment]
Good to go, thank you @dat-gcio!

**Comment by shiragoodman** [comment]
@f1337 this is the Staging ticket... did you intend to post ^ on the privacy & security ticket?

**Comment by joshkimux** [comment]
@shiragoodman @tchrisdyer is there a staging link ready for review? 🙇

**Comment by shiragoodman** [comment]
thanks for posting @joshkimux . @tchrisdyer if you don't have a  Staging Link, we will have to postpone this review. Please provide ASAP. Thanks,

**Comment by tchrisdyer** [comment]
Hi @shira, my understanding is that we're waiting on approval from @bkjohnson . If we get the approval before 2pm ET, we can get it pushed to Staging today. If not, my assumption is that it would be in staging tomorrow.

**Comment by shiragoodman** [comment]
@tchrisdyer we need all artifacts available, including the staging link, at least 4 days prior to the Staging Review. Unfortunately, I'm going to have to push this out to next Thursday. Our reviewers won't have adequate time to get their review done in time. Would 7/15 at 1:30pm work for you?

**Comment by tchrisdyer** [comment]
@shiragoodman yes, 7/15 at 1:30pm works for us

**Comment by shiragoodman** [comment]
thank you @tchrisdyer . Review has been rescheduled to 7/15 at 1:30pm ET

**Comment by dat-gcio** [comment]
Here are the links, and some hints to make sure you can get to the file upload compoonent.

Single file upload form:
1) Navigate to: https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/introduction
2) You only need to fill out primary or secondary caregiver, you do not have to fill out both
3) When you get to `Step 4 of 5: Representative document`, make sure you answer `Yes. I want to upload my document now. The next page will be the file upload.

Multiple file upload form:
1) Navigate to: https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction
2) Login. Credentials for vets.gov.user+33@gmail.com worked for me.
3) On `Step 3 of 5: Supporting evidence`, make sure to select `Yes` on the question `Is there any evidence you’d like us to review as part of your claim?`
4) On the question `What type of evidence do you want us to review as part of your claim?` select `Private medical records`
5) On the next page, select `yes` to the question `Do you want to upload your private medical records?` The next page will be the file upload

**Comment by shiragoodman** [comment]
@Dene-human thank you! Would you mind copy/pasting your last message into the main body of the ticket? This makes it easier for our reviewers to find everything they need (so they don't have to go digging through the comments). Thank you!

**Comment by dat-gcio** [comment]
@shiragoodman  updated!

**Comment by pjhill** [comment]
@dat-gcio , reading through your QA activities in the ticket I see the following:

> Product Test Cases: as this is a small incremental change to an existing message for an intermittent issue, we are primarily testing ad hoc.
Coverage for Reference: as this is a small incremental change to an existing message, no major adjustments being made

Did you discover any defects during your testing efforts? Do you have a list of those issues if they exist? Additionally, ad hoc testing can be great! We do encourage you all to use the exploratory testing / ad hoc testing template available in TestRail to record your objectives and results.
![image](https://user-images.githubusercontent.com/52456633/125108855-94c90500-e0b0-11eb-9843-489a80f4910b.png)

An template of an ad hoc / exploratory testing charter in markdown is available [here ](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/quality-assurance/manual-testing/exploratory-charter.md)as well.

That said, your comment [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26697#issuecomment-876557842), look like pretty great candidates for test cases that verify this change! By formalizing these scripts as test cases, you can reuse them to verify that your subsequent changes do not cause regressions as well.

**Comment by dat-gcio** [comment]
@pjhill I tested in the review instance during development, and found no defects there. I will test again in the staging environment with some documentation built out in testrail. The testing charter template is to be used for making a ticket correct?

**Comment by pjhill** [comment]
> @pjhill I tested in the review instance during development, and found no defects there. I will test again in the staging environment with some documentation built out in testrail. The testing charter template is to be used for making a ticket correct?

The charter there is a template intended to guide an ad hoc test session. You can drop that template into a ticket if you'd like, or you could add a copy of the template to your product folder here in va.gov-team and fill it out as you go.  Either way, thanks for your response!

**Comment by dat-gcio** [comment]
@pjhill Added test cases links to the main body of the ticket. I had to add the 21-526EZ under the 1010CG since our team doesn't own that form. I tested it to be thorough and it had multiple files capabilities.

[21-526EZ](https://dsvavsp.testrail.io/index.php?/cases/view/5515), [1010CG]

**Comment by allison0034** [comment]
@tchrisdyer I am getting a weird upload error. Should this be showing something different? 
![Screen Shot 2021-07-12 at 11 22 18 AM](https://user-images.githubusercontent.com/72398073/125313794-963e3b80-e303-11eb-8ba7-e73a2b835be0.png)

**Comment by allison0034** [comment]
@tchrisdyer I got the same error when I tried to upload a file bigger than 10 mg. Does staging currently have the changes?
![Screen Shot 2021-07-12 at 11 34 04 AM](https://user-images.githubusercontent.com/72398073/125315565-4496b080-e305-11eb-9b3a-7c8fce7e066a.png)

**Comment by tchrisdyer** [comment]
@dat-gcio can you take a look at @allison0034 's issues. She's getting some weird error messages on the file upload that I haven't seen before.

**Comment by dat-gcio** [comment]
@allison0034 I was able upload on the 1010CG. Have you tried to clear cache? Ctrl+Shift+R usual does the trick.

![image](https://user-images.githubusercontent.com/83595736/125316779-6fcdcf80-e306-11eb-865c-f4cbef5f2b98.png)

**Comment by allison0034** [comment]
@dat-gcio this would be for the error messages for a file bigger than 10 mg and a non supported file extension

**Comment by dat-gcio** [comment]
@allison0034 The error messages in your screenshots indicates that it came from the server. The work done was on the frontend error validation. From your screenshot, it looks like the files somehow made it through all the frontend checks. This is possible because anyone can change the extension of a file they possess. From searching your error, it seems like it comes from some security we have on va.gov. Did it not go away after clearing cache and cookies?

A wrong file file type should look something like this:
![image](https://user-images.githubusercontent.com/83595736/125320562-d7d1e500-e309-11eb-9479-6cecc428af7a.png)

And a too big file looks something like this:
![image](https://user-images.githubusercontent.com/83595736/125321650-ea005300-e30a-11eb-981b-31e52c221641.png)

**Comment by joshkimux** [comment]
@tchrisdyer for future staging reviews with sign in users, please use this [accessible passwords template](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-accessible-example.md) 🙇

**Comment by joshkimux** [comment]
@dat-gcio came across this error when attempting to go through the multiple upload flow. Would it be possible to fix this? 🙏 🙇 
<img width="1059" alt="Screen Shot 2021-07-13 at 2 19 53 PM" src="https://user-images.githubusercontent.com/14154792/125504719-3a9b6b1c-578f-4757-9934-caf675d854f5.png">

**Comment by dat-gcio** [comment]
@joshkimux It looks like there is something going on with a service that that form uses. I was able to find another form with multiple file upload.

https://staging.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/introduction
User vets.gov.user+503@id.me from:
https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-appeals.md

**Comment by peggygannon** [comment]
Hi, @dat-gcio I'm having the same issue as Allison. I'm only seeing the Invalid Authenticity Token error message.

**Comment by dat-gcio** [comment]
@peggygannon I tried to use a desktop virtual machine on the VA network, but its still working for me. Also had another team member test it was well, and they are only network error, not what you had. What browser are you using? Are you connected to SOCKS? Trying to figure out what the issue is.

Also are you able test with a file too large, too small, or wrong file extension at all? Those errors shouldn't make a network request. That was the goal of this change. We know that there are issues like these, so we changed the workflow on the front-end to make uploading after an error easier. We didn't make any back-end changes.

**Comment by dat-gcio** [comment]
@peggygannon @allison0034 I was able to get that error when I opened a new incognito tab and start filling out the 1010cg form. Created this [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/27273) to investigate it some more with my backend engineer. 

If you both had the issue due to incognito, I was able to get pass it by logging through id.me. If you were not in incognito, please let me know so we can expand our search for a fix.

**Comment by allison0034** [comment]
@dat-gcio I just tried an incognito window for the 2nd time and still have the same error. We do not have socks. I am using this link

**Comment by allison0034** [comment]
@dat-gcio I tried the incognito window yesterday and it did not work, however it I just tried and it seems to be working. I was able to get lost connection, wrong file type and file too large

**Comment by dat-gcio** [comment]
@allison0034 Good to hear that you were able to get through! I am trying to find the root cause at the moment, and it seems to be only happening when I start a fresh incognito session and go straight to the form. When you originally got that error, was it not in incognito?

**Comment by shiragoodman** [comment]
hi @tchrisdyer, @Dene-human, @dat-gcio and @mlighdc ,

As stated on the call, we are asking you to please complete a survey for the pilot of our new standards and Staging Review process. The survey link can be found here: https://adhoc.optimalworkshop.com/questions/xv854bro

We will be reviewing all feedback received from VFS teams, and will be making enhancements to our process prior to launch.

Thank you again!

@tchrisdyer  could you please forward the survey link to anyone I may have missed?
